server/src/main/java/org/candlepin/model/OwnerProductCurator.java

```diff
@@ -0,0 +1,400 @@
+        int count = 0;
+
+        for (Owner owner : owners) {
+            if (this.mapProductToOwner(product, owner)) {
+                ++count;
+            }
+        }
+
+        return count;
+    }
+
+    @Transactional
+    public int mapOwnerToProducts(Owner owner, Product... products) {
+        int count = 0;
+
+        for (Product product : products) {
+            if (this.mapProductToOwner(product, owner)) {
+                ++count;
+            }
+        }
+
+        return count;
+    }
+
+    @Transactional
+    public boolean removeOwnerFromProduct(Product product, Owner owner) {
+        String jpql = "DELETE FROM OwnerProduct op " +
+            "WHERE op.product.uuid = :product_uuid AND op.owner.id = :owner_id";
+
+        int rows = this.getEntityManager()
+            .createQuery(jpql)
+            .setParameter("owner_id", owner.getId())
+            .setParameter("product_uuid", product.getUuid())
+            .executeUpdate();
+
+        return rows > 0;
+    }
+
+    @Transactional
+    public int clearOwnersForProduct(Product product) {
+        String jpql = "DELETE FROM OwnerProduct op " +
+            "WHERE op.product.uuid = :product_uuid";
+
+        return this.getEntityManager()
+            .createQuery(jpql)
+            .setParameter("product_uuid", product.getUuid())
+            .executeUpdate();
+    }
+
+    @Transactional
+    public int clearProductsForOwner(Owner owner) {
+        String jpql = "DELETE FROM OwnerProduct op " +
+            "WHERE op.owner.id = :owner_id";
+
+        return this.getEntityManager()
+            .createQuery(jpql)
+            .setParameter("owner_id", owner.getId())
+            .executeUpdate();
+    }
+
+    /**
+     * Updates the product references currently pointing to the original product to instead point to
+     * the updated product for the specified owners.
+     * <p/></p>
+     * <strong>Warning:</strong> Hibernate does not gracefully handle situations where the data
+     * backing an entity changes via direct SQL or other outside influence. While, logically, a
+     * refresh on the entity should resolve any divergence, in many cases it does not or causes
+     * errors. As such, whenever this method is called, any active ActivationKey or Pool entities
+     * should be manually evicted from the session and re-queried to ensure they will not clobber
+     * the changes made by this method on persist, nor trigger any errors on refresh.
+     *
+     * @param current
+     *  The current product other objects are referencing
+     *
+     * @param updated
+     *  The product other objects should reference
+     *
+     * @param owners
+     *  A collection of owners for which to apply the reference changes
+     *
+     * @return
+     *  a reference to the updated product
+     */
+    public Product updateOwnerProductReferences(Product current, Product updated, Collection<Owner> owners) {
```

Comment:
Why is the third parameter a collection? The only usage of updateOwnerProductReferences that I see will always pass only single owner.